### PR TITLE
Add basic table population class

### LINKED ISSUE #3: create and test table filling functionality
Should make a `TemplateFiller` class (or similar) that initialises reading in a document.  Then a `populate_table` method that has a table number, and a pandas dataframe (that matches the dimensions of the output table), writing the them to the class's document. Also tests to ensure that the correct table is populated. 

prototype stub for using docx to write to rows.

```python
from pathlib import Path

from docx import Document
from docx.table import Table


def remove_all_rows(table: Table):
    """Slightly painful way of removing all rows except for title, but does work"""
    total_rows = len(table.rows) - 1
    for row_number in range(total_rows, 0, -1):
        row = table.rows[row_number]
        row._element.getparent().remove(row._element)


if __name__ == "__main__":
    template_path = Path("templates", "Reading Recovery School Report template 2021-22.docx")
    doc = Document(template_path)
    # You can get the tables from the document alone
    tables = doc.tables
    table_1 = tables[1]

    # view the header text, but can't read the comments in with python-docx
    header_cells = table_1.row_cells(0)
    print([cell.text.strip() for cell in header_cells])

    # Remove the rows that aren't the header
    print("rows:", [x for x in table_1.rows])
    remove_all_rows(table_1)
    print("post: deletion:", [x for x in table_1.rows])

    # Add a new row and populate the text value
    new_row = table_1.add_row()
    cell_count = 0
    for cell in new_row.cells:
        cell.text = f"{cell_count}: long text values wrap as expected based on their width"
        cell_count += 1
    print([cell.text.strip() for cell in table_1.row_cells(1)])

    doc.save(Path("output", "demo.docx"))
```

### PR DESCRIPTION
**Pull request by harryjmoss**
- Adds `TemplateFiller` class and tests to populate existing tables within a template MS Word (`.docx`) file.
- Adds a template `.docx` file to `input/templates` based off of the RRED report template file

Conscious that I might have added `filler.py` to the wrong directory (maybe should be one level up?) so happy to shift things around. 

Will close #3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -144,7 +144,8 @@ src/*/_version.py
 .idea/
 .vscode/
 
-input/*
 !input/*/.gitkeep
-output/*
+!input/templates/RRED_Report_Template_Single_Category.docx
```
Comment (stefpiatek) [inline .gitignore:148]: ooh maybe this file is better in`tests/resources/input...`?

File: src/rred_reports/reports/filler.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+from pathlib import Path
+
+import pandas as pd
+from docx import Document
+from docx.table import Table
+
+
+class TemplateFillerException(Exception):
```
Comment (stefpiatek) [inline src/rred_reports/reports/filler.py:8]: chef's kiss

File: src/rred_reports/reports/filler.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    def remove_all_rows(table: Table):
+        """Remove all rows in a table except for title
+
+        Args:
+            table (Table): Table within a .docx file
+        """
+        total_rows = len(table.rows) - 1
+        for row_number in range(total_rows, 0, -1):
+            row = table.rows[row_number]
+            row._element.getparent().remove(row._element)  # pylint: disable=protected-access
```
Comment (stefpiatek) [inline src/rred_reports/reports/filler.py:39]: yeah I couldn't find a public way to do it, and it seems the most sane way to approach it

File: src/rred_reports/reports/filler.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    def view_header(self, table: Table) -> list[Table.row_cells]:
```
Comment (stefpiatek) [inline src/rred_reports/reports/filler.py:41]: could also be a `staticmethod`, though is this only used in testing?

**Review by stefpiatek — APPROVED** [review]
🎉 love it, thanks Harry.

Generally non-blocking suggestions, though think maybe having specific test resources will make life easier on the .gitignore side of things. 

Adding in a helper test function to ensure all cells are filled in all tables of a word document would also be a nice touch for further development to use

**Comment by stefpiatek** [comment]
> Conscious that I might have added `filler.py` to the wrong directory (maybe should be one level up?) so happy to shift things around.

Forgot to answer this, nah I think this is a good place for it, we'll only be filling tables for reports so seems like a good place for it

File: .gitignore (hunk 1 above)
Comment (harryjmoss) [reply .gitignore:148]: Yep agreed, will move that.

File: src/rred_reports/reports/filler.py (hunk 4 above)
Comment (harryjmoss) [reply src/rred_reports/reports/filler.py:41]: Yeah only used for testing, bit guilty of adding this in as a future thing that might be nice to have. Happy to either make it a static method or just chop it entirely.

Comment (harryjmoss) [reply src/rred_reports/reports/filler.py]: Turns out I'm using this now so I'll keep it in!

**Pull request merged**